src/Sparrow.Server/LowMemory/MemoryInformation.cs

```diff
@@ -524,14 +608,59 @@ private static unsafe MemoryInfoResult GetMemoryInfoWindows(Process process, boo
             var fetchedInstalledMemory = GetPhysicallyInstalledSystemMemory(out var installedMemoryInKb);
 
             var sharedCleanInBytes = GetSharedCleanInBytes(process);
-            var availableMemoryForProcessingInBytes = (long)memoryStatus.ullAvailPhys + sharedCleanInBytes;
+            long memoryStatusUllAvailPhys = (long)memoryStatus.ullAvailPhys;
+            long totalPageFile = (long)memoryStatus.ullTotalPageFile;
+            long availPageFile = (long)(memoryStatus.ullTotalPageFile - memoryStatus.ullAvailPageFile);
+            var availableMemoryForProcessingInBytes = memoryStatusUllAvailPhys + sharedCleanInBytes;
+
+            string remarks = null;
+            JOBOBJECT_EXTENDED_LIMIT_INFORMATION limits = default;
+            if (QueryInformationJobObject(IntPtr.Zero, 
+                    JOBOBJECTINFOCLASS.ExtendedLimitInformation, (void*)&limits, 
+                    sizeof(JOBOBJECT_EXTENDED_LIMIT_INFORMATION),
+                    out int limitsOutputSize) == false || 
+                limitsOutputSize != sizeof(JOBOBJECT_EXTENDED_LIMIT_INFORMATION))
+            {
+                if (Logger.IsInfoEnabled)
+                    Logger.Info($"Failure when trying to query job object information info from Windows, error code is: {Marshal.GetLastWin32Error()}. Output size: {limitsOutputSize} instead of {sizeof(JOBOBJECT_EXTENDED_LIMIT_INFORMATION)}!");
+            }
+            else
+            {
+                long maxSize = long.MaxValue;
+                if (limits.BasicLimitInformation.MaximumWorkingSetSize != UIntPtr.Zero)
+                {
+                    maxSize = (long)limits.BasicLimitInformation.MaximumWorkingSetSize;
+                }
+
+                if (limits.ProcessMemoryLimit != UIntPtr.Zero)
+                {
+                    maxSize = Math.Min(maxSize, (long)limits.ProcessMemoryLimit);
+                }
+                
+                if (limits.JobMemoryLimit != UIntPtr.Zero)
+                {
+                    maxSize = Math.Min(maxSize, (long)limits.ProcessMemoryLimit);
+                }
+
+                if (maxSize != long.MaxValue)
+                {
+                    long workingSet64 = Process.GetCurrentProcess().WorkingSet64;
+                    availableMemoryForProcessingInBytes = maxSize - workingSet64;
```

Comment:
Potentially this can result in negative number, can't it?

Reply:
@ayende ?

Reply:
I don't believe so, no. That means that the process memory size is larger than the limit, and that should not be possible.

Reply:
I'm fine with doing `Math.Max(0,);`, though

Reply:
We'll be returning the limited available amount to `MemoryInfoResult` but I don't think this will guarantee that the process won't exceed that limit in some nasty case. Anyway, to be on the safe side I opened the following PR: https://github.com/ravendb/ravendb/pull/12554